hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java

```diff
@@ -87,7 +92,12 @@
     // operations, which meant that the index did not include some data.
     protected transient List<MapIndexInfo> mapIndexInfos;
 
-    private MapReplicationOperation operation;
+    // mapName -> null = full sync required
+    // mapName -> int[0] = no difference
+    // mapName -> int[2*n] = mapName -> Merkle tree node order/value pairs
+    protected Map<String, int[]> merkleTreeDiffByMapName = Collections.emptyMap();
```

Comment:
can we move all `merkleTreeDiffByMapName` related logic to EE to keep OS isolated? maybe we can introduce new methods in OS to override them in EE?